api/lds.proto

```diff
@@ -96,50 +126,54 @@ message FilterChain {
-  // via the LDS APIs.
+  // The unique name by which this listener is known. If no name is provided,
+  // Envoy will allocate an internal UUID for the listener. If the listener is to be dynamically
+  // updated or removed via :ref:`LDS <config_listeners_lds>` a unique name must be provided.
+  // By default, the maximum length of a listener's name is limited to 60 characters. This limit can be
+  // increased by setting the :option:`--max-obj-name-len` command line argument to the desired value.
   string name = 1;
 
   // The address that the listener should listen on.
-  Address address = 2;
+  Address address = 2 [(validate.rules).message.required = true];
 
-  // A list of filter chains to consider for this listener. The FilterChain with
-  // the most specific FilterChainMatch criteria is used on a connection. The
-  // algorithm works as follows:
+  // A list of filter chains to consider for this listener. The
+  // :ref:`FilterChain <envoy_api_msg_FilterChain>` with the most specific :ref:`FilterChainMatch
+  // <envoy_api_msg_FilterChainMatch>` criteria is used on a connection.
+  //
+  // .. attention::
   //
-  // 1. If SNI information is presented at connection time, only the
-  //    FilterChains matching the SNI are considered. Otherwise, only
-  //    FilterChains with no SNI domains are considered.
-  // 2. Of the FilterChains from step 1, the longest prefix match on the
-  //    bound destination address is used to select the next set of
-  //    FilterChains. This may be one FilterChain or multiple if there is
-  //    a tie.
-  // 3. The longest suffix match on the bound destination address is used to
-  //    select the FilterChain from step 2 that is used.
-  repeated FilterChain filter_chains = 3;
-
-  // If a connection is redirected using iptables, the port on which the proxy
-  // receives it might be different from the original destination port. When
-  // this flag is set to true, the listener uses the original destination
-  // address and port during FilterChain matching. Default is false.
+  //   In the current version, multiple filter chains are supported **only** so that SNI can be
+  //   configured. See the :ref:`FAQ entry <faq_how_to_setup_sni>` on how to configure SNI for more
+  //   information. When multiple filter chains are configured, each filter chain must have an
+  //   **identical** set of :ref:`filters <envoy_api_field_FilterChain.filters>`. If the filters
+  //   differ, the configuration will fail to load. In the future, this limitation will be relaxed such that
+  //   different filters can be used depending on which filter chain matches (based on SNI or
+  //   some other parameter).
+  repeated FilterChain filter_chains = 3 [(validate.rules).repeated.min_items = 1];
+
+  // If a connection is redirected using *iptables*, the port on which the proxy
+  // receives it might be different from the original destination port. When this flag is set to true,
```

Comment:
shouldn't this be about address, not just port? there's also fallback happening to 0.0.0.0:PORT as well as exact original IP:PORT match.